charms/istio-pilot/src/charm.py

```diff
@@ -130,6 +131,15 @@ def __init__(self, *args):
         #   application goes away we need to remove their resources
         self.framework.observe(self.on["ingress"].relation_changed, self.reconcile)
         self.framework.observe(self.on["ingress"].relation_broken, self.reconcile)
+
+        # Watch:
+        # * relation_created: because incomplete relation data should cause removal of the gateway
+        # * relation_joined: because incomplete relation data should cause removal of the gateway
```

Comment:
Is this right?

Reply:
I think so - what part are you thinking seems off?  

The worry here is that if someone relates to ingress-auth but the application they relate with doesn't send the required data, the user would think have enabled authentication on the ingress when they haven't.  To avoid this, we react on any relation event and either:
* if the relation data is correct: create the envoyfilter and enforce the authentication
* if the relation data is incorrect or incomplete (maybe before the related app sends us data, or if the related app messes up and sends nothing), remove the gateway so there is no traffic through the ingress and report the error

This way at worst a misconfiguration blocks traffic rather than leaving something that should be secure insecure.  

What do you think?

Reply:
@ca-scribner can you please add this explanation in the doc string? 
I believe that is very valuable for future us

Reply:
Added a comment to clarify this in the code.  This detail will also be in the design spec